harvester/harvest.py

```diff
@@ -197,8 +199,8 @@ def compare(self) -> None:
         logger.info("comparing our records with theirs")
 
         try:
-            external_ids = set(self.external_records.keys())
-            internal_ids = set(self.internal_records.keys())
+            external_ids = IndexedSet(self.external_records.keys())
+            internal_ids = IndexedSet(self.internal_records.keys())
```

Comment:
why do we need an IndexedSet?

Reply:
to guarantee insertion order. 

i found it necessary to make the tests in here pass reliably, then considered that we want sources to harvest in the same order every time.